client/src/components/Endpoints/getMinimalIcon.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {
+  AzureMinimalistIcon,
+  OpenAIMinimalistIcon,
+  ChatGPTMinimalistIcon,
+  PluginMinimalistIcon,
+  BingAIMinimalistIcon,
+  PaLMinimalistIcon,
+  AnthropicMinimalistIcon,
+} from '~/components/svg';
+import { useAuthContext } from '~/hooks';
+import { cn } from '~/utils';
+
+// Define the Props interface
+interface IconProps {
+  size?: number;
+  isCreatedByUser: boolean;
+  button?: boolean;
+  model?: string;
+  message?: boolean;
+  className?: string;
+  endpoint?: string | null;
+  chatGptLabel?: string;
+  modelLabel?: string;
+  jailbreak?: boolean;
+  error?: boolean;
+}
+
+const getMinimalIcon: React.FC<IconProps> = (props) => {
+  const { size = 30, isCreatedByUser, message = true } = props;
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { user } = useAuthContext();
+
+  if (isCreatedByUser) {
+    return (
+      <div
+        title={user?.name || 'User'}
+        style={{
+          width: size,
+          height: size,
+        }}
+        className={'relative flex items-center justify-center' + props?.className}
+      >
+        <img
+          className="rounded-sm"
+          src={
+            user?.avatar ||
+            `https://api.dicebear.com/6.x/initials/svg?seed=${
+              user?.name || 'User'
+            }&fontFamily=Verdana&fontSize=36`
+          }
+          alt="avatar"
+        />
+      </div>
+    );
+  } else if (!isCreatedByUser) {
+    const { endpoint, error } = props;
+
+    let icon, name;
+    if (endpoint === 'azureOpenAI') {
+      const { chatGptLabel } = props;
```

Comment:
repetitive code deconstructing the same variables. I would prefer you not use the chained if/else-if statements in favor of a map. see client/src/components/Input/ModelSelect/ModelSelect.tsx for an example